src/util/helpers.py

```diff
@@ -214,3 +216,94 @@ def _get_theme_models():
         "theme.website.menu": "website.menu",
         "theme.ir.attachment": "ir.attachment",
     }
+
+
+ResolvedFieldsPath = namedtuple(
+    "ResolvedFieldsPath",
+    "model path part_index field_model field_name relation_model",
+)
+
+
+def resolve_model_fields_path(cr, models_paths=None, input_paths_query=None, extra_where=None, extra_params=None):
```

Comment:
Okay. I'll simplify it like that. At some point I was already refactoring it like you said but I decided I really wanted also the ir.exports helpers to generate just one query. But we'll likely only have a handful of those kind of records, so it's premature optimization :see_no_evil:

Reply:
@aj-fuentes I think accepting multiple model_paths still makes sense though. So I'd keep the first argument as it is now.
E.g. in the ir.exports example I would call this with all the data from that table, do 1 query, it returns all the resolved paths, and filter the results back in the caller to match what I need. Otherwise I'd have to call this in a loop, which, while possibly negligible performance-wise, is still bad practice.

Reply:
> Otherwise I'd have to call this in a loop, which, while possibly negligible performance-wise, is still bad practice.

To me is worse practice to have overly complex util. If you show that there is a real performance impact then I will 100% agree with the more complex solution. Note: most of the upgrade time is spent in way more complex processing than updating/checking field paths, you may prove me wrong anyway :)

Reply:
Simplified. But I left it in `helpers.py`. Let me know if that has to change too.